vcd/resource_vcd_lb_service_monitor.go

```diff
@@ -0,0 +1,335 @@
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	lbMonitor, err := getLBMonitorType(d)
+	if err != nil {
+		return fmt.Errorf("unable to expand load balancer service monitor: %s", err)
+	}
+
+	createdMonitor, err := edgeGateway.CreateLBServiceMonitor(lbMonitor)
+	if err != nil {
+		return fmt.Errorf("error creating new load balancer service monitor: %s", err)
+	}
+
+	d.SetId(createdMonitor.ID)
+	return nil
+}
+
+func resourceVcdLbServiceMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBMonitor, err := edgeGateway.ReadLBServiceMonitor(&types.LBMonitor{ID: d.Id()})
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("unable to find load balancer service monitor with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBMonitorData(d, readLBMonitor)
+}
+
+func resourceVcdLbServiceMonitorUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	updateLBMonitorConfig, err := getLBMonitorType(d)
+	if err != nil {
+		return fmt.Errorf("could not expand monitor for update: %s", err)
+	}
+
+	updatedLBMonitor, err := edgeGateway.UpdateLBServiceMonitor(updateLBMonitorConfig)
+	if err != nil {
+		return fmt.Errorf("unable to update load balancer service monitor with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBMonitorData(d, updatedLBMonitor)
+}
+
+func resourceVcdLbServiceMonitorDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	err = edgeGateway.DeleteLBServiceMonitor(&types.LBMonitor{ID: d.Id()})
+	if err != nil {
+		return fmt.Errorf("error deleting load balancer service monitor: %s", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+// resourceVcdLbServiceMonitorImport expects dot formatted path to Load Balancer Service Monitor
+// i.e. my-org.my-org-vdc.my-edge-gw.my-lb-service-monitor
+func resourceVcdLbServiceMonitorImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified in such way my-org.my-org-vdc.my-edge-gw.my-lb-service-monitor")
```

Comment:
Suggestion: make the error message shorter and easier to read: "resource name must be specified as org.VDC.edge-gw.lb-service-monitor"

Reply:
Done.